### PR TITLE
Added Documentation & changed task-git to task-git-clone

### PR DESCRIPTION
**Pull request by Aneesh-M-Bhat**

### CODE REVIEW

File: docs/task-git-clone.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+```yaml
+---
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: example-taskrun
+spec:
+  taskRef:
+    name: git-cli
```
Comment (vdemeester) [inline docs/task-git-clone.md:15]: Should probably be `git-clone` 👼🏼

File: docs/task-git-clone.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+Please consider the [Workspaces](#workspaces), [Parameters](#parameters) and [Results](#results) described below.
+
+# Workspaces
+
+A single Workspace is required for this Task, while the optional Workspaces will allow advanced Git configuration and authentication.
+
+## `output`
+
+The `output` is a required Workspace, represents the primary location where the Git repository data will be stored, and later on, this Workspace will be monted in subsequent Pipeline's Task.
```
Comment (savitaashture) [inline docs/task-git-clone.md:32]:
```suggestion
The `output` is a required Workspace, represents the primary location where the Git repository data will be stored, and later on, this Workspace will be mounted in subsequent Pipeline's Task.
```

**Review by vdemeester — APPROVED** [review]
cc @savitaashture

**Comment by vdemeester** [comment]
/retest

🤔 openshift-ci is flaky these days…

**Pull request merged**